edb/edgeql/compiler/expr.py

```diff
@@ -1189,3 +1218,39 @@ def try_constant_set(expr: irast.Base) -> Optional[irast.ConstantSet]:
         )
     else:
         return None
+
+
+class IdentCompletionException(BaseException):
+    """An exception that is raised to halt the compilation and return a list of
+    suggested idents to be used at the location of qlast.Cursor node.
+    """
+
+    def __init__(self, suggestions: list[str]):
+        self.suggestions = suggestions
+
+
+@dispatch.compile.register(qlast.Cursor)
+def compile_Cursor(
```

Comment:
Some other things worth considering support for: module names, object names inside a module once a module is specified, functions